ext/ImplicitDifferentiationChainRulesExt.jl

```diff
@@ -25,57 +26,53 @@ function ChainRulesCore.rrule(
     rc::RuleConfig,
     implicit::ImplicitFunction,
     x::AbstractArray{R},
-    ::Val{return_byproduct};
+    ::ReturnByproduct;
     kwargs...,
-) where {R,return_byproduct}
-    @unpack conditions, linear_solver, presolver = implicit
+) where {R}
+    @unpack conditions, linear_solver = implicit
 
-    y, z = implicit(x, Val(true); kwargs...)
+    y, z = implicit(x, ReturnByproduct(); kwargs...)
     n, m = length(x), length(y)
 
     backend = ReverseRuleConfigBackend(rc)
     pbA = pullback_function(backend, _y -> conditions(x, _y, z; kwargs...), y)
     pbB = pullback_function(backend, _x -> conditions(_x, y, z; kwargs...), x)
     pbmA = PullbackMul!(pbA, size(y))
     pbmB = PullbackMul!(pbB, size(y))
+
     Aᵀ_op = LinearOperator(R, m, m, false, false, pbmA)
     Bᵀ_op = LinearOperator(R, n, m, false, false, pbmB)
-    implicit_pullback = ImplicitPullback(
-        presolver(Aᵀ_op, x, y), Bᵀ_op, linear_solver, x, Val(return_byproduct)
-    )
+    Aᵀ_op_presolved = presolve(linear_solver, Aᵀ_op, y)
+
+    implicit_pullback = ImplicitPullback(Aᵀ_op_presolved, Bᵀ_op, linear_solver, x)
```

Comment:
this pullback should return 3 outputs but it should only return 2 when no `ReturnByProduct` is input.

Reply:
actually it should always return 2 outputs, one for the function itself and one for x

idk what the third one was doing there